### DISCUSSION TITLE
Difficulty Configuring Domain Username in Teleport Role

### DISCUSSION THREAD

**Discussion by zelda001** [gravitational/teleport#40386] [Q&A] [answered]
Hello, I am trying to configure a user role in Teleport that allows access to Windows desktops using domain usernames. I have included the domain prefix in the windows_desktop_logins field of the YAML configuration (e.g., nameofsystem\Username1). However, this approach hasn't been successful. Unfortunately, I can't login to this resource using "Username1" alone as its giving me multiple errors.  

![Screenshot 2024-04-10 155101](https://github.com/gravitational/teleport/assets/113965976/ade49977-19f6-4868-a0a6-5fb0c7614108)

![Screenshot 2024-04-10 155140](https://github.com/gravitational/teleport/assets/113965976/b0bef99f-5e4c-4e3b-b755-ef5b0acab134)

Is there specific YAML syntax to include the domain name, or are there alternative methods if domain usernames aren't directly supported?

Any advice would be great. Thank you very much

**Comment by zmb3** [comment]
There is no need to include the domain in the allowed usernames.

When you connect to the desktop, the domain will automatically be added if the desktop is registered with Teleport as belonging to an AD domain.

**Reply by zelda001** [reply]
Is it using this configuration https://goteleport.com/docs/desktop-access/active-directory-manual/? Unfortunately, Teleport talks through a secure LDAP, which we can't do at the moment. 

If its not that, how do I register the desktop with Teleport as belonging to an AD domain? Thanks

**Reply by zmb3** [reply, marked as answer]
There are two options to connect to Windows hosts with RDP:

1) Connect as domain users. This requires AD with a proper PKI setup and secure LDAP.
2) Connect as local users. This requires a separate piece of software you install on each Windows host.

Sounds like you want domain users, which rules out option 2 completely.
It also sounds like you can't do secure LDAP, which rules out option 1.
We're now out of options unfortunately 😢

**Reply by zelda001** [reply]
We were already considering migrating to secure LDAP in the future, so we might speed up that process so we can set up Teleport correctly. Thank you very much :)
